java-checks/src/main/java/org/sonar/java/checks/NullPointerCheck.java

```diff
@@ -72,13 +76,26 @@ public void visitMemberSelectExpression(MemberSelectExpressionTree tree) {
     super.visitMemberSelectExpression(tree);
   }
 
+  @Override
+  public void visitMethodInvocation(MethodInvocationTree tree) {
+    Symbol symbol = ((MethodInvocationTreeImpl) tree).getSymbol();
+    if (symbol.isKind(Symbol.MTH)) {
+      MethodSymbol methodSymbol = (MethodSymbol) ((MethodInvocationTreeImpl) tree).getSymbol();
+      for (int i = 0; i < tree.arguments().size(); i += 1) {
+        List<Symbol> parameters = methodSymbol.getParameters().scopeSymbols();
+        if (!canBeNull(parameters.get(i < parameters.size() ? i : parameters.size() - 1))) {
```

Comment:
I would probably not rely on a ternary operator in this case, just to make it clear for further reading that you are doing so to handle problems of varargs (at the very least a small commentary could be nice)